### ISSUE TITLE
Add API version number to 404 page or add an endpoint /api-version

### ISSUE THREAD

**Issue by IsmailM** [phenopolis/phenopolis_browser#291]
Could use semantic versioning or simply some sort of git commit id.

**Comment by pontikos** [comment]
This will be implemented by default when we move to FastAPI I asssume @dvarrazzo

**Comment by IsmailM** [comment]
At the moment we show the following on 404:

```
{
"success": false,
"error": {
"type": "NotFound",
"message": []
},
"remote_addr": "https://api-dev.phenopolis.org",
"full_path": "/?",
"method": "GET",
"scheme": "https",
"timestamp": "[2021-May-19 11:35]"
}```

So should be easy to add the version number if we store it somewhere...

**Comment by alanwilter** [comment]
Question, when I do: http://localhost:8888/abc It takes me to `The page you were looking for doesn't exist.`. Looking at the code in `frontend/src/page/NotFoundPage.js` I see:

```javascript
export default function NoFoundPage() {
  return (
```
Did you note **`NoFoundPage`**? Nowhere in the project I found this `NoFoundPage`, but instead I see several references to  `NotFoundPage` (did you note the **Not**?), like in `frontend/src/App.js`:

```javascript
import NotFoundPage from './page/NotFoundPage';
```
I've changed `NoFoundPage` to `NotFoundPage` in my local test but didn't notice any difference, I don't know how serious is it. @YuanTian1991, do you think it is a real error?

I'm just trying to find a way to "tag" our dev-live version for debugging etc.

**Comment by alanwilter** [comment]
Ideally, I'd like to have a place, e.g., in `public.env`, to set a var called, say, `INT_VERSION` with a tag.
The tag could be the latest commit id but I can't see how to get it since commit sha-1 depends on the file being changed itself.

I looked at `.gitattributes` but I still don't figure out how we could use it.

My main line now is to use `git hooks`. I'm reading more about it. Looking at `pre-commit` but check this first: https://stackoverflow.com/a/49383779/2534342

We could have a `pre-commit` hook that would:
1- create a value (for simplicity, I'd use e.g. `date +%Y%m%d_%H%M%S`);
2- update this value in `public.env`;
3- automatically add `public.env` with whatever others files are being committed.

Then inside our code we could use `dotenv` to get this `INT_VERSION` tag and use in 404 report, when sending emails with 4xx or 5xx errors, in api_version endpoint etc. Even fronted could pickup this value and show it in the footnote.

Another potential issue issue with git hooks is how do they work on Windows? I guess everybody here uses either Linux or Mac but I'd still be concerned.

This used to be so simple with Subversion.

**Comment by YuanTian1991** [comment]
Hi @alanwilter The name should not matter, you can name it anything after import.

**Comment by alanwilter** [comment]
Thanks @YuanTian1991, I ended up committing `NotFoundPage` not intentionally (it went by mistake in one of my commits), but thankfully it's not harmful.

**Comment by alanwilter** [comment]
It's now fixed in commit 3278be7
